packages/test/test-end-to-end-tests/src/test/sharedTreeEndToEndTests.spec.ts

```diff
@@ -0,0 +1,461 @@
+		// won't make a difference. So, only run them for local server to reduce the number of test combinations
+		// it runs in.
+		if (provider.driver.type !== "local") {
+			this.skip();
+		}
+
+		// Create a loader and a detached container.
+		const loader1 = provider.createLoader([[provider.defaultCodeDetails, runtimeFactory]]);
+		container1 = await loader1.createDetachedContainer(provider.defaultCodeDetails);
+		// Get the create new request to attach the container with.
+		const request = provider.driver.createCreateNewRequest(provider.documentId);
+		dataObject1 = (await container1.getEntryPoint()) as DataObjectWithTree;
+		tree1 = dataObject1.tree;
+		await container1.attach(request);
+
+		// Create a second loader and load a second container from the first container's URL.
+		const loader2 = provider.createLoader([[provider.defaultCodeDetails, runtimeFactory]]);
+		const loadUrl = await container1.getAbsoluteUrl("");
+		assert(loadUrl !== undefined, "Container's absolute URL is undefined");
+		container2 = await loader2.resolve({ url: loadUrl });
+		const dataObject2 = (await container2.getEntryPoint()) as DataObjectWithTree;
+		tree2 = dataObject2.tree;
+	});
+
+	const sf = new SchemaFactory("sharedTreeE2ETests");
+	// Simple schema for a point object with x and y coordinates.
+	class Point extends sf.object("Point", {
+		x: sf.number,
+		y: sf.number,
+	}) {}
+
+	describe("Tree with a simple object", () => {
+		const treeViewConfig = new TreeViewConfiguration({ schema: Point });
+		let initialPoint: Point;
+		let treeViewClient1: TreeView<typeof Point>;
+		let treeViewClient2: TreeView<typeof Point>;
+
+		beforeEach(async () => {
+			treeViewClient1 = tree1.viewWith(treeViewConfig);
+			treeViewClient2 = tree2.viewWith(treeViewConfig);
+			initialPoint = new Point({ x: 1, y: 2 });
+			treeViewClient1.initialize(initialPoint);
+			await provider.ensureSynchronized();
+		});
+
+		it("can sync initial tree data", async () => {
+			// Validate that the second client received the initial tree data.
+			assert.deepStrictEqual(
+				treeViewClient2.root,
+				initialPoint,
+				"Initial tree data not synchronized",
+			);
+		});
+
+		it("can sync updates to tree data", async () => {
+			// Make changes to the x in first client and y in second client.
+			treeViewClient1.root.x = 3;
+			treeViewClient2.root.y = 4;
+			await provider.ensureSynchronized();
+
+			// Validate that the changes are synchronized.
+			assert.strictEqual(
+				treeViewClient2.root.x,
+				3,
+				"Changes via first client not synchronized",
+			);
+			assert.strictEqual(
+				treeViewClient1.root.y,
+				4,
+				"Changes via second client not synchronized",
+			);
+		});
+	});
+
+	// A slightly complex schema for a shape object that contains an array of points.
+	class Shape extends sf.object("Shape", {
+		sides: sf.number,
+		points: sf.array(Point),
+	}) {}
+
+	function addSide(treeView: TreeView<typeof Shape>, point: Point) {
```

Comment:
I think this would make way more sense as a method on the shape rather than a free method taking in a view.

Reply:
Same for removeSide below